server/data/welcomeClient.ts

```diff
@@ -120,11 +122,17 @@ export default class WelcomeClient {
     }) as Promise<TemporaryAbsence>
   }
 
-  async confirmTemporaryAbsence(prisonNumber: string, prisonId: string): Promise<ArrivalResponse | null> {
+  async confirmTemporaryAbsence(
+    prisonNumber: string,
+    prisonId: string,
+    arrivalId?: string
+  ): Promise<ArrivalResponse | null> {
     logger.info(`welcomeApi: confirmTemporaryAbsence ${prisonNumber})`)
     try {
       return (await this.restClient.post({
-        path: `/temporary-absences/${prisonNumber}/confirm`,
+        path: arrivalId
+          ? `/temporary-absences/${prisonNumber}/confirm?arrivalId=${arrivalId}`
+          : `/temporary-absences/${prisonNumber}/confirm`,
```

Comment:
Not a huge fan of these ternary expressions - I was thinking of building a ResourceBuilder class, where we could build up a URL with optional query parameters dynamically.

Think it'd be needed if we start adding more than the one query parameter, but maybe leave for it for now?